main.rb

```diff
@@ -1,17 +1,140 @@
 module Enumerable
-    def my_each
-        for i in 0...self.size
-            yield(self[i])
-        end
-    self
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    i = 0
+    if block_given?
+      while i < length
+        yield(self[i])
+        i += 1
+      end
+      return self
+    end
+    to_enum
+  end
+
+  def my_each_with_index
+    i = 0
+    if block_given?
+      while i <= length - 1
+        yield(self[i], i)
+        i += 1
+      end
+      return self
     end
+    to_enum
+  end
+
+  def my_select
+    arr = []
+    return to_enum unless block_given?
+
+    my_each { |x| arr << x if yield(x) == true }
+    arr
+  end
 
-    def my_each_with_index
-        for i in 0...self.size
-            yield(self[i], i)
+  def my_all?(arg = nil, &prc)
+    if block_given?
+      my_each { |x| return false if prc.call(x) == false }
+    elsif arg.is_a?(Regexp)
+      my_each { |x| return false if arg.match?(x.to_s) == false }
+    elsif arg.is_a?(Class)
+      my_each { |x| return false if x.is_a?(arg) == false }
+    elsif arg.nil? == false
+      my_each { |x| return false if x != arg }
+    else
+      my_each { |x| return false if x.nil? == true }
+    end
+    true
+  end
+
+  def my_any?(&prc)
+    if block_given?
+      my_each { |x| return true if prc.call(x) }
+    else
+      my_each { |x| return true if x.nil? == false }
+    end
+    false
+  end
```

Comment:
I don't think that you have covered the cases added in the previous comment. Let me put them again->
`my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
[1, 2, 3].my_any? #should return true
[1, 2, nil].my_any? #should return true
[false, false, nil].my_any? #should return false
```
- when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
[1, 2, 3].my_any?(Integer) #should return true
[1, 'demo', false].my_any?(Integer) #should return true
['demo', false, nil].my_any?(Integer) #should return false
```
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_any?(/t/)    #should return false.
%w[dog door rod blade].my_any?(/d/)   #should return true.
```
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
[1, 2, 3].my_any?(3)   #should return true
[1, 2, 2].my_any?(3)   #should return false
```